test/linalg/generic.jl

```diff
@@ -200,6 +199,16 @@ let
     @test LinAlg.axpy!(α, x, deepcopy(y)) != Matrix{Int}[α] .* x
 end
 
+# test that LinAlg.axpy! works for x and y of different dimensions
+let
+    α = 5
+    x = ones(Int, 2, 3)
```

Comment:
probably use `x = 2:7` or similar, so that the test checks that the right elements of `x` are being added to `y`.